### ISSUE TITLE
PLR0133: False Positive

### ISSUE THREAD

**Issue by colin99d** [astral-sh/ruff#11232]
* A minimal code snippet that reproduces the bug.
```
x = 5
x or 0 > 0
```
* The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.
`ruff check .`
* The current Ruff version (`ruff --version`).
`ruff 0.4.2`

**Comment by MichaReiser** [comment]
It's not clear to me why you consider this a false positive. The `0 > 0` is a constant comparison. Could you mind explaining your reasoning in more detail?

**Comment by colin99d** [comment]
According to the ruff docs, the reason for this rule is that:

`Comparing two constants will always resolve to the same value, so the comparison is redundant. Instead, the expression should be replaced with the result of the comparison.`

However, given that x is of type None | int, then this will not always yield the same result. In our code, we often use or statements to handle None values. So the usage is basically "if x is None, compare with 0 instead". Given that x or 0 is not a constant, I don't think this rule is helpful here.

**Comment by trag1c** [comment]
@colin99d You're not taking operator precedence into account here; `x or 0 > 0` is correctly flagged as it's actually `x or (0 > 0)`, as `>` has higher precedence than `or` (see https://docs.python.org/3.12/reference/expressions.html#operator-precedence).

**Comment by colin99d** [comment]
Oh wow nice catch @trag1c. Will close.
